src/core/load_balancing/rls/rls.cc

```diff
@@ -808,7 +827,10 @@ void RlsLb::ChildPolicyWrapper::Orphaned() {
   GRPC_TRACE_LOG(rls_lb, INFO)
       << "[rlslb " << lb_policy_.get() << "] ChildPolicyWrapper=" << this
       << " [" << target_ << "]: shutdown";
-  is_shutdown_ = true;
+  {
+    MutexLock lock(&lb_policy_->mu_);
+    is_shutdown_ = true;
```

Comment:
Why do we need to re-acquire the lock here?  I think that once we remove the references to the `ChildPolicyWrapper` from the cache entry, the picker can't use them, so I'm not sure we need to hold the lock when setting `is_shutdown_`.

Reply:
It's being read here - https://github.com/grpc/grpc/blob/fe249ceb258bf57681782f3c46fbfb40bf12de34/src/core/load_balancing/rls/rls.cc#L939 after we take a lock on the mutex. Is it protected by WorkSerializer instead?

Reply:
Yes, that will always be called inside the WorkSerializer.

Reply:
Got it, I removed it outside the mutex and added a comment that it's protected by the WorkSerializer

Reply:
The code here is still grabbing the mutex unnecessarily to modify `is_shutdown_`.

Reply:
ofcourse i removed the original one but not the one i added :D